pandas/tests/frame/methods/test_first_and_last.py

```diff
@@ -95,3 +99,48 @@ def test_empty_not_input(self, func):
         result = getattr(df, func)(offset=1)
         tm.assert_frame_equal(df, result)
         assert df is not result
+
+    @pytest.mark.parametrize("start, periods", [("2010-03-31", 1), ("2010-03-30", 2)])
+    def test_first_with_first_day_last_of_m_DO(self, frame_or_series, start, periods):
+        x = frame_or_series([1] * 100, index=bdate_range(start, periods=100))
+        result = x.first(pd.DateOffset(days=periods))
+        expected = frame_or_series(
+            [1] * periods, index=bdate_range(start, periods=periods)
+        )
+        tm.assert_equal(result, expected)
+
+    def test_first_with_first_day_end_of_frq_n_greater_one_DO(self, frame_or_series):
+        x = frame_or_series([1] * 100, index=bdate_range("2010-03-31", periods=100))
+        result = x.first(pd.DateOffset(days=2))
+        expected = frame_or_series(
+            [1] * 2, index=bdate_range("2010-03-31", "2010-04-01")
+        )
+        tm.assert_equal(result, expected)
+
+    def test_first_w_DateOffset(self):
+        # GH#51284
+        i = pd.date_range("2018-04-09", periods=4, freq="2D")
+        x = DataFrame({"A": [1, 2, 3, 4]}, index=i)
+        result = x.first(pd.DateOffset(days=3))
+        expected = DataFrame(
+            {"A": [1, 2]}, index=pd.date_range("2018-04-09", periods=2, freq="2D")
+        )
+        tm.assert_equal(result, expected)
+
+    def test_first_w_DateOffset_other(self):
+        # GH#45908
+        i = pd.date_range("2018-04-09", periods=30, freq="2D")
+        x = DataFrame({"A": Series(np.arange(30), index=i)}, index=i)
+        result = x.first(pd.DateOffset(days=15))
+        i2 = pd.date_range("2018-04-09", periods=8, freq="2D")
+        expected = DataFrame({"A": Series(np.arange(8), index=i2)}, index=i2)
```

Comment:
would just
```
expected = DataFrame({"A": Series(np.arange(8), index=i2)}, index=i2)
```
work the same? fine to keep the input as in the original post, but for `expected` let's try to be as simple as possible

Reply:
Actually, I don't see a difference on line 136 and what is  written on the comment.
I assume that I should do more like:
```
expected = DataFrame({"A": np.arange(8)}, index=i2)
```

Reply:
yup, that's right!